docs/source/overview/deployment.rst

```diff
@@ -0,0 +1,183 @@
+Cloud Custodian.
+
+.. _compliance_as_code:
+
+Compliance as Code
+------------------
+When operating Cloud Custodian, it is highly recommended to treat the policy
+files as code, similar to that of Terraform or CloudFormation files. Cloud
+Custodian has a built-in dryrun mode and policy syntax validation which when
+paired with an automated CI system, can help you release policies with confidence.
+
+This tutorial assumes that you have working knowledge of Github, Git, Docker,
+and a continuous integration tool (Jenkins, Drone, Travis, etc.).
+
+To begin, start by checking your policy files into a source control management
+tool like Github. This allows us to version and enable collaboration through
+git pull requests and issues. In this example, we will be setting up a new repo
+in Github.
+
+First, set up a new repo in Github and grab the reporistory url. You don't need
+need to add a README or any other files to it first.
+
+.. code-block:: bash
+
+    $ mkdir my-policies 
+    $ cd my-policies
+    $ git init
+    $ git remote add origin <github repo url>
+    $ touch policy.yml
+
+Next, we'll add a policy to our new ``policy.yml`` file.
+
+.. code-block:: yaml
+
+    policies:
+      - name: aws-vpcs
+        resource: aws.vpc
+
+Once you've added the policy to your policy file we can stage our changes from our
+working directory and push it up to our remote:
+
+.. code-block:: bash
+
+    # this should show your policy.yml as an untracked file
+    $ git status
+
+    $ git add policy.yml
+    $ git commit -m 'init my first policy'
+    $ git push -u origin master
+
+Once you've pushed your changes you should be able to see your new changes inside
+of Github. Congratulations, you're now ready to start automatically validating and
+testing your policies!
+
+.. _continuous_integration_of_policies:
+
+Continuous Integration of Policies
+----------------------------------
+
+Next, enable a CI webhook back to your CI system of choice when pull requests
+targeting your master branch are opened or updated. This allows us to constantly
+test and validate the policies that are being modified. You can use the tool
+c7n_policystream to generate diffs between the current master version and the
+proposed change in the pull request.
+
+In this example, we will be using Microsoft Azure Devops Pipelines.
+
+First, navigate to https://azure.microsoft.com/en-us/services/devops/pipelines/ and
+click the "Start pipelines free with Github" button and follow the flow to connect
+your Github account with Devops Pipelines.
+
+Next click on the Piplines section in the left hand side of the sidebar and connect
+with Github. Once the pipeline is setup, we can add the following azure devops
+configuration to our repo:
+
+.. code-block:: yaml
+
+    trigger:
+    - master
+    
+    jobs: 
+      - job: 'Validate'
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        strategy:
+          matrix:
```

Comment:
don't really need a matrix strategy here, we're just using a single python version